### ISSUE TITLE
Improved TS types for event handlers

### ISSUE THREAD

**Issue by DamonOehlman** [GetStream/stream-chat-js#263]
Currently, the event registration and deregistration methods (`on` and `off` respectively) aren't correctly typed for registration of the "catch all" event handler case, i.e. it's not possible to do the following and have TypeScript be happy with you:

```ts
client.on(() => {
  console.log('got an event');
});
```

This can be pretty easily fixed by changing the type definitions to support the overloaded forms of the methods.  For example the type definitions on the client would look something like:

```ts
on(eventName: string, eventHandler: any): { unsubscribe(): void };
on(eventHandler: any): { unsubscribe(): void };
off(eventName: string, eventHandler: any): void;
off(eventHandler: any): void;
```

You could actually go pretty hardcore on this typing if you ever feel so inclined, with strong definitions for specific type definitions for each of the different event and details with regards to the data that is sent through from event calls. e.g.

```ts
type EventRegistrationResult = { unsubscribe: () => void };
type MessageNewHandler = (evt: MessageNewEvent) => void;
type HealthCheckHandler = (evt: HealthCheckEvent) => void;

on(eventName: 'message.new', eventHandler: MessageNewHandler): EventRegistrationResult;
on(eventName: 'health.check', eventHandler: HealthCheckHandler): EventRegistrationResult;
...
on(eventHandler: GlobalEventHandler): EventRegistrationResult;
```

That's pretty extreme, but I'm sure all of us using TypeScript would be over the moon if we ended up there one day :)

More than happy to raise a PR to take care of the simple case of handling the global event handler if that's helpful.

**Comment by vishalnarkhede** [comment]
Hey @DamonOehlman fixed in `v1.6.0`. Please reopen if it doesn't work for you!

**Comment by DamonOehlman** [comment]
@vishalnarkhede That's fantastic news - thanks heaps :)
